wrappers/azurekeyvault/azurekeyvault.go

```diff
@@ -287,31 +353,80 @@ func (v *Wrapper) buildBaseURL() string {
 }
 
-func (v *Wrapper) getKeyVaultClient(withCertPool *x509.CertPool) (*azkeys.Client, error) {
-	var err error
-	var cred azcore.TokenCredential
+func (v *Wrapper) getCredential() (azcore.TokenCredential, error) {
+	var (
+		cred azcore.TokenCredential
+		err  error
+	)
 
-	switch {
-	// Use a fully specified tenant, client, secret if provided
-	case v.tenantID != "" && v.clientID != "" && v.clientSecret != "":
-		cred, err = azidentity.NewClientSecretCredential(v.tenantID, v.clientID, v.clientSecret, nil)
+	switch v.authMethod {
+	case DefaultAzureCredential:
+		cred, err = azidentity.NewDefaultAzureCredential(nil)
 		if err != nil {
-			return nil, fmt.Errorf("failed to get client secret credentials %w", err)
+			return nil, fmt.Errorf("failed to acquire default identity credentials: %w", err)
 		}
-	case v.clientID != "":
-		// Try a managed service credential with a specified client id
-		clientID := azidentity.ClientID(v.clientID)
+	case EnvironmentCredential:
+		cred, err = azidentity.NewEnvironmentCredential(nil)
+		if err != nil {
+			return nil, fmt.Errorf("failed to acquire environment credentials: %w", err)
+		}
+	case ManagedIdentityCredential:
 		cred, err = azidentity.NewManagedIdentityCredential(&azidentity.ManagedIdentityCredentialOptions{
-			ID: clientID,
+			ID: azidentity.ClientID(v.clientID),
 		})
 		if err != nil {
 			return nil, fmt.Errorf("failed to get managed identity credentials: %w", err)
 		}
-	// By default let Azure select existing credentials
+	case ClientSecretCredential:
+		cred, err = azidentity.NewClientSecretCredential(v.tenantID, v.clientID, v.clientSecret, nil)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get client secret credentials: %w", err)
+		}
+	case CertificateCredential:
+		certData, err := os.ReadFile(v.certPath)
+		if err != nil {
+			return nil, fmt.Errorf(`failed to read certificate file "%s": %v`, v.certPath, err)
+		}
+
+		var password []byte
+		if v.certPass != "" {
+			password = []byte(v.certPath)
+		}
+		certs, key, err := azidentity.ParseCertificates(certData, password)
+
+		cred, err = azidentity.NewClientCertificateCredential(v.tenantID, v.clientID, certs, key, nil)
+	case WorkloadIdentityCredential:
+		cred, err = azidentity.NewWorkloadIdentityCredential(nil)
+	case Automatic:
 	default:
-		cred, err = azidentity.NewDefaultAzureCredential(nil)
+		return nil, fmt.Errorf("unknown authentication method")
+	}
+
+	return cred, nil
+}
+
+func (v *Wrapper) getAutomaticCredential() (azcore.TokenCredential, error) {
+	switch {
+	case v.tenantID != "" && v.clientID != "" && v.clientSecret != "":
+		return azidentity.NewClientSecretCredential(v.tenantID, v.clientID, v.clientSecret, nil)
+	case v.clientID != "":
+		return azidentity.NewManagedIdentityCredential(&azidentity.ManagedIdentityCredentialOptions{
+			ID: azidentity.ClientID(v.clientID),
+		})
+	default:
+		return azidentity.NewDefaultAzureCredential(nil)
+	}
```

Comment:
Ah sorry, I think I described this incorrectly. I was thinking something like:

```suggestion
	switch v.authMethod {
	case Automatic:
		switch {
		case v.tenantID != "" && v.clientID != "" && v.clientSecret != "":
			return v.getCredential(ClientSecretCredential)
		case v.clientID != "":
			return v.getCredential(ManagedIdentityCredential)
		default:
			return v.getCredential(v.DefaultAzureCredential)
		}
	default:
		return v.getCredential(v.authMethod)
	}
```